torch/csrc/jit/interpreter.cpp

```diff
@@ -400,26 +400,31 @@ struct AutogradHandle : public ContainerTensor {
 // behave differently depending on whether a future handle needs to be
 // created.
 struct HandleBuilder {
-  HandleBuilder(bool requires_handle) {
+  HandleBuilder(bool requires_handle, bool values_are_variables)
+    : values_are_variables(values_are_variables) {
     if(requires_handle) {
       handle = new AutogradHandle();
       handle->forward_inputs = std::make_shared<DummyFunction>();
     }
   }
   autograd::Variable addInput(at::Tensor && input, const VariableFlags & flags_) {
-    if(handle && flags_.requires_grad) {
-      auto variable = autograd::make_variable(std::move(input), /*requires_grad=*/false);
+    if (handle && flags_.requires_grad) {
+      auto variable = values_are_variables ?
+        static_cast<autograd::Variable&&>(input).detach() :
```

Comment:
This won't work correctly with the GraphExecutor pipeline because doesn't support handles correctly and detach will break the autograd gradients (in fact, we really should start removing the Handle functionality entirely).  When we are tracing using `torch.jit.trace`, which will turn into a GraphExecutor, we should be creating python functions with no handle outputs, and with  `tracing_autograd_python_function=false`.

Reply:
Right, I forgot that the Eval/Handle path is broken. I'll update the PR to stop tracing Python functions this way and remove the handles. They make the code very confusing.